Automattic/wp-calypso
# Components: add pagination control to component package

## Changes proposed in this Pull Request
Currently, there are two ( [DotPager](https://github.com/Automattic/wp-calypso/blob/e2e9876c56a38ba83fb5d1a002bb117bd7c6aebf/client/components/dot-pager/index.jsx) and [Welcome Guide](https://github.com/Automattic/wp-calypso/blob/trunk/apps/editing-toolkit/editing-toolkit-plugin/wpcom-block-editor-nux/src/welcome-tour/pagination.js) ) instances of "dot" pagination, this PR creates a component that can be used in both instances and potentially more in the future. This is loosely based off the [Gutenberg Guide component](https://github.com/WordPress/gutenberg/blob/4bbfbc13603d28fcc45368c529954164bf8581de/packages/components/src/guide/page-control.js).

This is a direct replacement for [the Welcome Guide pagination](https://github.com/Automattic/wp-calypso/blob/trunk/apps/editing-toolkit/editing-toolkit-plugin/wpcom-block-editor-nux/src/welcome-tour/pagination.js) and so it also removes the current `pagination.js` in favor of this new package.

Originally I planned on incorporating the buttons into the component but that seemed a bit too complex. Instead, I allowed the buttons to be passed in as children. The children are wrapped in the `li` and are pushed to the opposite side of the box.

## Screenshots
![Screen Capture on 2022-01-24 at 15-33-14](https://user-images.githubusercontent.com/33258733/150802995-f2639ad3-c31f-45fa-964f-8cf44cc54469.gif)

## Testing instructions

1. Pull branch and navigate to `apps/editing-toolkit/`
2. Run `yarn dev --sync` 
3. Log in to your sandbox and open your sandboxed site.
4. View the Welcome Guide from the editor and verify the dots are working.
5. Change the color scheme of your WP-ADMIN area and verify the colors update as well.

Related to #59757
